ticktock-jvm/tzdb/src/main/java/dev/zacsweers/ticktock/jvm/tzdb/JvmTzdbZoneRules.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (C) 2020 Zac Sweers & Gabriel Ittner
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package dev.zacsweers.ticktock.jvm.tzdb;
+
+import dev.zacsweers.ticktock.runtime.TickTockPlugins;
+import dev.zacsweers.ticktock.runtime.TzdbZoneDataProvider;
+
+/** Entry point for lazy zone rules on JVM. */
+public final class JvmTzdbZoneRules {
+
+  /** Initializes the TickTockZoneRulesProvider using a {@link TzdbZoneDataProvider}. */
+  public static void init() {
+    System.setProperty(
+        "java.time.zone.DefaultZoneRulesProvider",
+        "dev.zacsweers.ticktock.runtime.TickTockZoneRulesProvider");
+    TickTockPlugins.setZoneDataProvider(TzdbZoneDataProvider::create);
```

Comment:
You need to call both `TickTockPlugins.setZoneDataProvider` and `TickTockPlugins.setZoneIdsProvider`. Like here
https://github.com/ZacSweers/ticktock/blob/45bb16b0db7d5371688c30b9ac612452221ea3f8/ticktock-android/tzdb/src/main/java/dev/zacsweers/ticktock/android/tzdb/AndroidTzdbZoneRules.java#L37-L43

Reply:
If we change all Supplier usages from `Supplier<...Provider` to `Supplier<? extends ...Provider>`  we can also get rid of the two lambdas there and just pass the `supplier` instance directly. I forgot that in my pr.